x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -26,7 +27,60 @@
 import static org.elasticsearch.xpack.esql.formatter.TextFormat.CSV;
 import static org.elasticsearch.xpack.esql.formatter.TextFormat.URL_PARAM_DELIMITER;
 
+/**
+ * Listens for a single {@link EsqlQueryResponse}, builds a corresponding {@link RestResponse} and sends it.
+ */
 public class EsqlResponseListener extends RestResponseListener<EsqlQueryResponse> {
+    /**
+     * A simple, thread-safe stop watch for timing a single action.
+     * Allows to stop the time for building a response and to e.g. log it at a later point.
+     */
+    private static class ThreadSafeStopWatch {
+        /**
+         * Start time of the watch
+         */
+        private long startTimeNS;
+
+        /**
+         * End time of the watch
+         */
+        private long endTimeNS;
+
+        /**
+         * Is the stop watch currently running?
+         */
+        private boolean running;
+
+        /**
+         * Start the stop watch. Will do nothing if it is already running.
+         */
+        public void start() {
+            synchronized (this) {
```

Comment:
We can remove the start method and make the startTimeNS field final and initialize it in the constructor.